ui/app/components/calendar-widget.js

```diff
@@ -3,139 +3,115 @@ import layout from '../templates/components/calendar-widget';
-    return this.displayYear === startYear; // if on startYear then don't let them click back to the year prior
+    // calendar widget should only go as far back as the passed in start time
+    return isSameYear(this.calendarDisplayDate, this.startDate);
   }
   get widgetMonths() {
-    const displayYear = this.displayYear;
-    const currentYear = this.currentYear;
-    const currentMonthIdx = this.currentMonth;
-    const [startMonth, startYear] = this.args.startTimeDisplay.split(' ');
-    const startMonthIdx = this.args.arrayOfMonths.indexOf(startMonth);
-    return this.args.arrayOfMonths.map((month, idx) => {
-      const monthId = `${idx}-${displayYear}`;
+    const startYear = this.startDate.getFullYear();
+    const startMonthIdx = this.startDate.getMonth();
+    return ARRAY_OF_MONTHS.map((month, index) => {
       let readonly = false;
 
       // if widget is showing billing start year, disable if month is before start month
-      if (parseInt(startYear) === displayYear && idx < startMonthIdx) {
+      if (startYear === this.getDisplayYear && index < startMonthIdx) {
         readonly = true;
       }
 
-      // if widget showing current year, disable if month is current or later
-      if (displayYear === currentYear && idx >= currentMonthIdx) {
+      // if widget showing current year, disable if month is later than current month
+      if (this.getDisplayYear === this.currentDate.getFullYear() && index > this.currentDate.getMonth()) {
         readonly = true;
       }
       return {
-        id: monthId,
-        month,
+        index,
+        year: this.getDisplayYear,
+        name: month,
         readonly,
-        current: monthId === `${currentMonthIdx}-${currentYear}`,
       };
     });
   }
 
-  // HELPER FUNCTIONS (alphabetically) //
-  addClass(element, classString) {
-    element.classList.add(classString);
-  }
-
-  removeClass(element, classString) {
-    element.classList.remove(classString);
-  }
-
-  resetDisplayYear() {
-    let setYear = this.currentYear;
-    if (this.args.endTimeDisplay) {
-      try {
-        const year = this.args.endTimeDisplay.split(' ')[1];
-        setYear = parseInt(year);
-      } catch (e) {
-        console.debug('Error resetting display year', e); // eslint-disable-line
-      }
-    }
-    this.displayYear = setYear;
-  }
-
   // ACTIONS (alphabetically) //
   @action
   addTooltip() {
     if (this.disablePastYear) {
-      let previousYear = Number(this.displayYear) - 1;
-      this.tooltipText = `${previousYear} is unavailable because it is before your billing start month. Change your billing start month to a date in ${previousYear} to see data for this year.`; // set tooltip text
+      const previousYear = Number(this.getDisplayYear) - 1;
+      this.tooltipText = `${previousYear} is unavailable because it is before your billing start date. Change your billing start month to a date in ${previousYear} to see data for this year.`;
```

Comment:
Might need to confer with design but can this message be made more generic and not reference billing?

Reply:
@ivana-mcconnell - any strong feelings on removing the word 'billing'. It'd make it more sensical for OSS users. 
<img width="300" alt="Screen Shot 2022-11-08 at 11 33 58 AM" src="https://user-images.githubusercontent.com/68122737/200658285-1110fa8f-877e-4a4c-b0f1-223001d0e0c6.png">